quickwit/quickwit-cli/src/index.rs

```diff
@@ -643,122 +645,222 @@ pub async fn describe_index_cli(args: DescribeIndexArgs) -> anyhow::Result<()> {
+    const LENGTH: usize = 7;
+
+    fn fields(&self) -> Vec<String> {
+        vec![
+            self.index_id.clone(),
+            self.index_uri.to_string(),
+            self.num_published_splits.to_string(),
+            self.num_published_docs.to_string(),
+            format!("{} MB", self.size_published_docs),
+            display_option_in_table(&self.timestamp_field_name),
+            display_timestamp_range(&self.timestamp_range),
+        ]
+    }
+
+    fn headers() -> Vec<String> {
+        vec![
+            "Index ID: ".to_string(),
+            "Index URI: ".to_string(),
+            "Number of published splits: ".to_string(),
+            "Number of published documents: ".to_string(),
+            "Size of published splits (MB): ".to_string(),
+            "Timestamp field: ".to_string(),
+            "Timestamp range: ".to_string(),
+        ]
+    }
+}
+
+fn display_option_in_table(opt: &Option<impl Display>) -> String {
+    match opt {
+        Some(opt_val) => format!("{}", opt_val),
+        None => "Field does not exist for the index.".to_string(),
+    }
+}
+
+fn display_timestamp_range(range: &Option<(Option<i64>, Option<i64>)>) -> String {
+    match range {
+        Some((Some(timestamp_min), Some(timestamp_max))) => {
+            format!("{} -> {}", timestamp_min, timestamp_max)
+        }
+        _ => "Range does not exist for the index.".to_string(),
+    }
+}
+
+impl IndexStats {
+    pub fn from_metadata(
+        index_metadata: IndexMetadata,
+        splits: Vec<Split>,
+    ) -> anyhow::Result<Self> {
+        let splits_num_docs = splits
             .iter()
-            .map(|split| split.split_metadata.time_range.clone())
-            .filter(|time_range| time_range.is_some())
-            .map(|time_range| *time_range.unwrap().start())
-            .min();
-        let time_max = splits
+            .map(|split| split.split_metadata.num_docs)
+            .sorted()
+            .collect_vec();
+
+        let total_num_docs = splits_num_docs.iter().sum::<usize>();
+
+        let splits_bytes = splits
             .iter()
-            .map(|split| split.split_metadata.time_range.clone())
-            .filter(|time_range| time_range.is_some())
-            .map(|time_range| *time_range.unwrap().start())
-            .max();
-        println!(
-            "{:<35} {:?} -> {:?}",
-            "Timestamp range:".color(GREEN_COLOR),
-            time_min,
-            time_max
-        );
+            .map(|split| (split.split_metadata.footer_offsets.end / 1_000_000) as usize)
+            .sorted()
+            .collect_vec();
+        let total_bytes = splits_bytes.iter().sum::<usize>();
+
+        let timestamp_range = if index_metadata.indexing_settings.timestamp_field.is_some() {
+            let time_min = splits
+                .iter()
+                .flat_map(|split| split.split_metadata.time_range.clone())
+                .map(|time_range| *time_range.start().min(time_range.end()))
```

Comment:
Why do we need `time_range.start().min(time_range.end()`? Is `time_range.start() <= time_range.end()` not always true?

Reply:
`start >= end` is possible for `RangeInclusive<T>` but it should not be possible for `SplitMetadata` you are right. Fixed.